code/app/src/main/java/com/example/the_tarlords/MainActivity.java

```diff
@@ -261,4 +287,95 @@ private String generateNewUserId() {
         return Settings.Secure.getString(getContentResolver(), Settings.Secure.ANDROID_ID);
     }
 
+    /**
```

Comment:
Does everything from line 291 onwards need to be in main activity?  I'm not sure if this code could go in the map fragment instead as I don't know exactly how it works, but I think it would probably make more sense there, or maybe as part of a separate map class? If it needs to be in an activity, there is also the possibility of a separate Maps activity that could open from the event details fragment. This is just me throwing around ideas though, if you think what I'm suggesting will be overly complicated then it's probably not worth it

Reply:
Also if it's specifically being used just to set user location on check in it could possibly go in QR activity since that's the only place check ins are actually handled (not sure if this would cause problems if a user enabled location services, checked in, then disabled location services though)